dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java

```diff
@@ -466,16 +466,19 @@ public void launchWithArgs(final String entry, final String localEntry, final St
                         final String descriptor = descriptorType.getValue().toLowerCase();
                         LanguageClientInterface languageClientInterface = convertCLIStringToEnum(descriptor);
                         DescriptorLanguage language = DescriptorLanguage.convertShortStringToEnum(descriptor);
-
+                        if (workflow.getDefaultTestParameterFilePath() == null) {
+                            conditionalErrorMessage((yamlRun != null) == (jsonRun != null),
+                                                    "The workflow does not specify a default test parameter file, use --json or --yaml to specify one",
+                                                    CLIENT_ERROR);
+                        }
                         try {
                             switch (language) {
                             case CWL:
-                                conditionalErrorMessage((yamlRun != null) == (jsonRun != null), "Either --json or --yaml is required", CLIENT_ERROR);
                                 languageClientInterface.launch(entry, false, yamlRun, jsonRun, null, uuid);
                                 break;
                             case WDL:
                             case NEXTFLOW:
-                                conditionalErrorMessage(jsonRun == null, "dockstore: missing required flag --json", CLIENT_ERROR);
+                                conditionalErrorMessage((yamlRun != null), "Nextflow only supports --json, --yaml cannot be used", CLIENT_ERROR);
```

Comment:
The `case` statements fall through, so this code is executed for both the `WDL` and `NEXTFLOW` cases.